### ISSUE TITLE
[SignalR cpp] Allow handler registrations while connection is active

### ISSUE THREAD

**Issue by BrennanConroy** [dotnet/aspnetcore#27445]
Currently `hubConnection.on(...)` will throw if the connection is active, this isn't the pattern we follow in the other clients and we should allow handlers to be registered while the connection is running.

https://github.com/aspnet/SignalR-Client-Cpp/blob/bcea0940576306ff625caa508b4f6a2a80d3fbe3/src/signalrclient/hub_connection_impl.cpp#L83-L86

**Comment by CodeMnke** [comment]
All you need to do is just remove the check, there is no reason for it.

**Comment by BrennanConroy** [comment]
> All you need to do is just remove the check

Not quite, the collection being inserting into is not thread safe so more thought needs to be put into the fix than just "remove the check".

If you're passionate about this issue the repo is open source, feel free to open a PR to fix it 😃
